src/admin/mod.rs

```diff
@@ -380,6 +383,57 @@ async fn handle_metrics(reg: Arc<Mutex<Registry>>, _req: Request<Body>) -> Respo
         .unwrap()
 }
 
+async fn handle_logging(req: Request<Body>) -> Response<Body> {
+    match *req.method() {
+        hyper::Method::GET => {
+            let loglevel = match telemetry::get_current() {
+                Ok(level) => format!("current log level is {}\n", level),
+                Err(e) => format!("failed to get log level, err: {}\n", e),
+            };
+            Response::builder()
+                .status(hyper::StatusCode::OK)
+                .header(hyper::header::CONTENT_TYPE, "text/plain")
+                .body(loglevel.into())
+                .unwrap()
+        }
+        hyper::Method::POST => match get_new_loglevel(req).await {
+            Ok(level) => match telemetry::set_level(level) {
+                Ok(return_str) => Response::builder()
+                    .status(hyper::StatusCode::OK)
+                    .header(hyper::header::CONTENT_TYPE, "text/plain")
+                    .body(format!("{}\n", return_str).into())
+                    .unwrap(),
+                Err(return_str) => Response::builder()
+                    .status(hyper::StatusCode::INTERNAL_SERVER_ERROR)
+                    .body(format!("failed to change log level, {}\n", return_str).into())
+                    .unwrap(),
+            },
+            Err(msg) => Response::builder()
+                .status(hyper::StatusCode::NOT_FOUND)
+                .body(format!("failed to get the new log level, {}\n", msg).into())
+                .unwrap(),
+        },
+        _ => Response::builder()
+            .status(hyper::StatusCode::METHOD_NOT_ALLOWED)
+            .body("Invalid HTTP method\n".to_string().into())
+            .unwrap(),
+    }
+}
+
+async fn get_new_loglevel(req: Request<Body>) -> Result<String, Error> {
+    let body = hyper::body::to_bytes(req).await?;
```

Comment:
confused why we are parsing the Body, shouldn't it be like `/logging?level=debug`?

Reply:
We should match https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging as closely as possible

Reply:
sure, John, I will make changes according to the doc, will get you noticed when it's ready, thanks for all of your comments, it's valuable for me.

Reply:
> We should match https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging as closely as possible



> We should match https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging as closely as possible

John, do you mind to check the comments to see if this is the correct behavior? thanks! I also refactored most of the code, regarding the fine-grained log level on single/multiple mods, that requires more effort, working on it.